File: 8kyu/8kyu - Is it a number.js
// 8kyu - Is it a number?

// Given a string s, write a method (function) that will return 
// true if its a valid single integer or floating number or false if its not.

// const isDigit = s => s == parseFloat(s)
// const isDigit = s => !!s.trim() && !isNaN(s)
// const isDigit = s => parseFloat(s) == Number(s)
// const isDigit = s => /^-?\d+(\.\d+)?$/.test(s)


q = isDigit("s2324") // false
q
q = isDigit("-234.4") // true
q
q = isDigit("3") // true
q
q = isDigit("  3  ") // true
q
q = isDigit("-3.23") // true
q
q = isDigit("3-4") // false
q
q = isDigit("  3   5") // false
q
q = isDigit("3 5") // false
q
q = isDigit("zero") // false
q

File: 8kyu/8kyu - Is it a number.py
# 8kyu - Is it a number?

""" Given a string s, write a method (function) that will return true if its a valid single integer or floating number or false if its not.

Valid examples, should return true:

isDigit("3")
isDigit("  3  ")
isDigit("-3.23")

should return false:

isDigit("3-4")
isDigit("  3   5")
isDigit("3 5")
isDigit("zero") """

# def isDigit(string):
#     try: return bool(float(string)+1)
#     except: return False


def isDigit(string):
    try: return bool(float(string)) or True
    except: return False


q = isDigit("s2324")  # False
q
q = isDigit("-234.4")  # True
q
